ballerina/utils.bal

```diff
@@ -188,6 +188,20 @@ isolated function generateCanonicalHeaders(map<string> headers, http:Request? re
     return [canonicalHeaders, signedHeaders];
 }
 
+# Function to generate signing key.
+#
+# + secretAccessKey - Value of the secret key
+# + shortDateStr - shortDateStr Parameter Description
+# + region - Endpoint region
+# + return - Signing key
+isolated function generateSigningKey(string secretAccessKey, string shortDateStr, string region) returns byte[]|error {
+    string signValue = AWS4 + secretAccessKey;
+    byte[] dateKey = check crypto:hmacSha256(shortDateStr.toBytes(), signValue.toBytes());
+    byte[] regionKey = check crypto:hmacSha256(region.toBytes(), dateKey);
+    byte[] serviceKey = check crypto:hmacSha256(SERVICE_NAME.toBytes(), regionKey);
+    return check crypto:hmacSha256(TERMINATION_STRING.toBytes(), serviceKey);
```

Comment:
```suggestion
    return crypto:hmacSha256(TERMINATION_STRING.toBytes(), serviceKey);
```

check will be an unnecessary extra check here since you're returning anyway.